python/server/qgsserverresponse.sip

```diff
@@ -0,0 +1,80 @@
+/***************************************************************************
+                          qgsserverresponse.sip
+
+  Define response class for services
+  -------------------
+  begin                : 2016-12-05
+  copyright            : (C) 2016 by David Marteau
+  email                : david dot marteau at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+/**
+ * \ingroup server
+ * QgsServerResponse
+ * Class defining response interface passed to services QgsService::executeRequest() method
+ */
+
+class QgsServerResponse
+{
+%TypeHeaderCode
+#include "qgsserverresponse.h"
+%End
+  public:
+
+    //!constructor
+    QgsServerResponse();
+
+    //! destructor
+    virtual ~QgsServerResponse();
+
+    /** Set Header entry
+     *  Add Header entry to the response
+     *  Note that it is usually an error to set Header after writing data
+     */
+    virtual void setHeader( const QString& key, const QString& value ) = 0;
+
+    /** Set the http return code
+     * @param code HTTP return code value
+     */
+    virtual void setReturnCode( int code ) = 0;
```

Comment:
HTTP return code is not necessary an error code: you can return code <> 200 with legitimate content. This is different that sendError( ) which delegate the error handling to the transport/IO framework. sendError() should end the transaction and any attempt to write data or set headers should be an error.  Maybe the comments should be completed.